src/api/app/services/trigger_controller_service/scm_extractor.rb

```diff
@@ -21,36 +21,62 @@ def call
     private
 
     def github_extractor_payload
-      {
+      payload = {
         scm: 'github',
-        commit_sha: @payload.dig(:pull_request, :head, :sha),
-        pr_number: @payload[:number],
-        source_branch: @payload.dig(:pull_request, :head, :ref),
-        target_branch: @payload.dig(:pull_request, :base, :ref),
-        action: @payload[:action], # TODO: Names may differ, maybe we need to find our own naming (defer to service?)
-        source_repository_full_name: @payload.dig(:pull_request, :head, :repo, :full_name),
-        target_repository_full_name: @payload.dig(:pull_request, :base, :repo, :full_name),
         event: @event,
         api_endpoint: github_api_endpoint
       }
+
+      case @event
+      when 'pull_request'
+        payload.merge!({ commit_sha: @payload.dig(:pull_request, :head, :sha),
+                         pr_number: @payload[:number],
+                         source_branch: @payload.dig(:pull_request, :head, :ref),
+                         target_branch: @payload.dig(:pull_request, :base, :ref),
+                         action: @payload[:action],
+                         source_repository_full_name: @payload.dig(:pull_request, :head, :repo, :full_name),
+                         target_repository_full_name: @payload.dig(:pull_request, :base, :repo, :full_name) })
+      when 'push'
+        payload.merge!({ commit_sha: @payload[:after],
+                         # We need this for Workflows::YAMLDownloader#download_url
+                         target_branch: @payload[:ref].sub('refs/heads/', ''),
+                         # We need this for Workflow::Step#branch_request_content_github
+                         source_repository_full_name: @payload.dig(:repository, :full_name),
+                         # We need this for SCMStatusReporter#call
+                         target_repository_full_name: @payload.dig(:repository, :full_name) })
```

Comment:
Is this correct? `source_repository_full_name` and `target_repository_full_name` are pointing towards same element?

Reply:
Yes, see the comments above each line.